src/data/equiv/ring.lean

```diff
@@ -220,11 +249,11 @@ variables [non_unital_non_assoc_semiring R] [non_unital_non_assoc_semiring S]
   (f : R ≃+* S) (x y : R)
 
 /-- A ring isomorphism sends zero to zero. -/
-@[simp] lemma map_zero : f 0 = 0 := (f : R ≃+ S).map_zero
+protected lemma map_zero : f 0 = 0 := map_zero f
 
 variable {x}
 
-@[simp] lemma map_eq_zero_iff : f x = 0 ↔ x = 0 := (f : R ≃+ S).map_eq_zero_iff
+protected lemma map_eq_zero_iff : f x = 0 ↔ x = 0 := map_eq_zero_iff f (equiv_like.injective f)
```

Comment:
Missing `has_pointwise_zero` typeclass? 😝 

Why unsimping it, though?

Reply:
Good point, we should have a `simp` version of `map_eq_zero_iff` specially for 0-preserving equivs. For now:
```suggestion
@[simp] lemma map_eq_zero_iff : f x = 0 ↔ x = 0 := map_eq_zero_iff f (equiv_like.injective f)
```

Reply:
We already have that. We we need is one for zero-preserving embeddings (or equivs directly).

Reply:
```suggestion
protected lemma map_eq_zero_iff : f x = 0 ↔ x = 0 := add_equiv_class.map_eq_zero_iff f
```